packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmDictionaryTests.kt

```diff
@@ -0,0 +1,2966 @@
+        } else {
+            DOUBLE_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        String::class -> if (nullable) {
+            NULLABLE_STRING_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            STRING_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        RealmInstant::class -> if (nullable) {
+            NULLABLE_TIMESTAMP_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            TIMESTAMP_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        ObjectId::class -> if (nullable) {
+            NULLABLE_OBJECT_ID_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            OBJECT_ID_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        BsonObjectId::class -> if (nullable) {
+            NULLABLE_BSON_OBJECT_ID_VALUES.mapIndexed { i, value ->
+                Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value)
+            }
+        } else {
+            BSON_OBJECT_ID_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        RealmUUID::class -> if (nullable) {
+            NULLABLE_UUID_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            UUID_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        ByteArray::class -> if (nullable) {
+            NULLABLE_BINARY_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            BINARY_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        RealmObject::class -> NULLABLE_DICTIONARY_OBJECT_VALUES.mapIndexed { i, value ->
+            Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value)
+        }
+        Decimal128::class -> if (nullable) {
+            NULLABLE_DECIMAL128_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS_FOR_NULLABLE[i], value) }
+        } else {
+            DECIMAL128_VALUES.mapIndexed { i, value -> Pair(DICTIONARY_KEYS[i], value) }
+        }
+        RealmAny::class -> {
+            val anyValues = REALM_ANY_PRIMITIVE_VALUES + REALM_ANY_REALM_OBJECT
+
+            // Generate as many keys as RealmAny values
+            var key = 'A'
+            val keys = anyValues.map { key.also { key += 1 } }
+
+            // Now create pairs of key-RealmAny for the dataset
+            anyValues.mapIndexed { i, value -> Pair(keys[i].toString(), value) }
+        }
+        else -> throw IllegalArgumentException("Wrong classifier: '$classifier'")
+    } as List<T>
+}
+
+/**
+ * Tester interface defining the operations that have to be tested exhaustively.
+ */
+internal interface DictionaryApiTester<T, Container> : ErrorCatcher {
+
+    val realm: Realm
+
+    override fun toString(): String
+    fun copyToRealm()
```

Comment:
Looks like we are missing testing for `equals/hashCode` for the `RealmDictionary` itself?

Reply:
True, I will add them 👍